shm/DryFruit.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef DRYFRUIT_HPP
+#define DRYFRUIT_HPP
+#include "Fruit.hpp"
+
+// Class used to represent cargo of subtype dry fruit type carried by merchant ship
+class DryFruit : public Fruit 
+{
+private:
+    size_t counter_{0};
+
+public:
+    using Fruit::Fruit;
+    size_t getPrice() const override
+    {
+        if (timeInCargoRoom_ >= shelfLife_)
+            return 0;
+        return static_cast<size_t>(
+            3 * basePrice_ * ((float)(shelfLife_ - timeInCargoRoom_)) / shelfLife_);
```

Comment:
try to be consistent. You use static_cast in C++ style, then (float) in C style, Don't use C style here.